builtin/providers/aws/resource_aws_security_group.go

```diff
@@ -407,6 +411,23 @@ func resourceAwsSecurityGroupUpdateRules(
 		ns := n.(*schema.Set)
 
 		remove := expandIPPerms(group, os.Difference(ns).List())
+
+		if len(remove) == 0 && len(os.List()) == 0 && ruleset == "egress" {
```

Comment:
`len(remove) == 0 && len(os.List()) == 0` may be redundant, but I wasn't sure if `len(os.List()) == 0` was enough. Should it be sufficient to check for `nil` here?